NINA.Plugin.Speckle/NINA.Plugin.Speckle/Speckle.cs

```diff
@@ -65,6 +75,63 @@ public override Task Teardown() {
             return base.Teardown();
         }
 
+        public ICommand OpenFileCommand { get; private set; }
+
+        public string ReferenceStarListLocation {
+            get => _pluginOptionsAccessor.GetValueString(nameof(ReferenceStarListLocation), "");
+            set {
+                _pluginOptionsAccessor.SetValueString(nameof(ReferenceStarListLocation), value);
+                RaisePropertyChanged();
+            }
+        }
+
+        private void OpenFile() {
+            OpenFileDialog fileDialog = new OpenFileDialog();
+            fileDialog.DefaultExt = ".csv"; // Required file extension 
+            fileDialog.Filter = "Csv documents (.csv)|*.csv"; // Optional file extensions
+
+            if (fileDialog.ShowDialog() == DialogResult.OK) {
+                ReferenceStarListLocation = fileDialog.FileName;
+            }
+        }
+
+        private AsyncObservableCollection<ReferenceStar> _referenceStarList;
+
+        [JsonProperty]
+        public AsyncObservableCollection<ReferenceStar> ReferenceStarList {
+            get => _referenceStarList;
+            set {
+                _referenceStarList = value;
+                RaisePropertyChanged();
+            }
+        }
+
+        private bool _LoadingReferenceStars = false;
+
+        public bool LoadingReferenceStars {
+            get { return _LoadingReferenceStars; }
+            set {
+                _LoadingReferenceStars = value;
+                RaisePropertyChanged();
+            }
+        }
+
+        public async Task LoadReferenceStarList() {
+            if (string.IsNullOrWhiteSpace(ReferenceStarListLocation) || LoadingReferenceStars) {
+                Logger.Debug("No path to reference star list.");
+                return;
+            }
```

Comment:
Add a check to make sure the file exists... `if (!File.Exists(ReferenceStarListLocation)) { complain() }`